packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -491,42 +476,44 @@ export class TSRHandler {
 	private _triggerUpdateDevices() {
 		if (!this._initialized) return
 
-		if (this._triggerUpdateDevicesTimeout) clearTimeout(this._triggerUpdateDevicesTimeout)
+		if (this._triggerUpdateDevicesTimeout) {
+			clearTimeout(this._triggerUpdateDevicesTimeout)
+		}
 		this._triggerUpdateDevicesTimeout = undefined
 
-		if (!this._updateDevicesIsRunning) {
-			this._updateDevicesIsRunning = true
-			debug('triggerUpdateDevices now')
-
-			// Defer:
-			setTimeout(() => {
-				this._updateDevices().then(
-					() => {
-						this._updateDevicesIsRunning = false
-						if (this._triggerUpdateDevicesCheckAgain) {
-							debug('triggerUpdateDevices from updateDevices promise resolved')
-							if (this._triggerUpdateDevicesTimeout) clearTimeout(this._triggerUpdateDevicesTimeout)
-							this._triggerUpdateDevicesTimeout = setTimeout(() => this._triggerUpdateDevices(), 1000)
-						}
-						this._triggerUpdateDevicesCheckAgain = false
-					},
-					() => {
-						this._updateDevicesIsRunning = false
-						if (this._triggerUpdateDevicesCheckAgain) {
-							debug('triggerUpdateDevices from updateDevices promise rejected')
-							setTimeout(() => this._triggerUpdateDevices(), 1000)
-							if (this._triggerUpdateDevicesTimeout) clearTimeout(this._triggerUpdateDevicesTimeout)
-							this._triggerUpdateDevicesTimeout = setTimeout(() => this._triggerUpdateDevices(), 1000)
-						}
-						this._triggerUpdateDevicesCheckAgain = false
-					}
-				)
-			}, 10)
-		} else {
-			// oh, it's already running, cue a check again later:
+		if (this._updateDevicesIsRunning) {
 			debug('triggerUpdateDevices already running, cue a check again later')
 			this._triggerUpdateDevicesCheckAgain = true
+			return
 		}
+		this._updateDevicesIsRunning = true
+		debug('triggerUpdateDevices now')
+
+		// Defer:
+		setTimeout(() => {
+			this._updateDevices()
+				.then(
+					() =>
+						this._triggerUpdateDevicesCheckAgain &&
```

Comment:
Agreed, I feel this makes sense in React, since otherwise, doing simple conditional switches makes stuff insanely verbose, but we should avoid it outside JSX.